packages/logo-grid/partials/dialog-tooltip/index.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useRef, useEffect } from 'react'
+import { DialogOverlay, DialogContent } from '@reach/dialog'
```

Comment:
I don't mean to throw a wrench in your progress here, but do you think https://www.npmjs.com/package/@reach/popover would be more appropriate for this use case?

Reply:
This one doesn't appear on their website 🤔

Reply:
Yeah, not sure why that is exactly. I do think it's the right component for this use though. It looks like they use it as part of their combobox component.

Reply:
Ah nice - yeah support for the `position` prop could definitely help clean things up a little and make this feel a little less like a re-implementation of Tippy. Hopefully it'll resolve the annoying `useRect` issue I was running into as well. Will give it a shot 👍

Reply:
@BRKalow I gave `@reach/popover` a shot - did up needing some workarounds, but now have both `DialogTooltip` and `PopoverTooltip` with identical APIs which we can choose from. A quick summary:

- `DialogTooltip` uses `@reach/dialog`, which is [well-documented](https://reach.tech/dialog), and aligns somewhat with reach.tech's recommendation to use `dialog` for "interactive tooltips". By using `dialog`, we get focus behaviour, keyboard and on-click-outside functionality to dismiss the dialog by default. But, the implementation required significant workarounds, including custom positioning, and a weird effect to force re-rendering whenever the dialog is opened or closed.

- `PopoverTooltip` uses `@reach/popover`. It seems to have no official documentation - unclear why this is. By using `popover`, we get focus behaviour and custom positioning by default. But, the implementation required the use of `onClickOutside` and `onFocusOutside` to allow the popover to be dismissed as expected when a visitor clicks or focuses away.

Leaning towards using `@reach/popover` as you recommended, wanted to run this by just in case. If that sounds good, will just commit the deletion of the `dialog`-based implementation and this should be good to go 👍 🙇

Reply:
Thanks for spiking this out. The popover implementation seems potentially cleaner...what do you think? It seems better to have to handle the click/focus outside than the custom positioning. I'll let you make the final call here though!

Reply:
+1 I think `popover` is a bit cleaner - will go with that unless @jescalan has any strong objections 👍